staging/src/k8s.io/kube-gen/cmd/conversion-gen/generators/conversion.go

```diff
@@ -338,36 +338,42 @@ func (e equalMemoryTypes) Equal(a, b *types.Type) bool {
 	if equal, ok := e[conversionPair{b, a}]; ok {
 		return equal
 	}
-	result := e.equal(a, b)
+	result := e.equal(a, b, currentlyBuildingTypes)
 	e[conversionPair{a, b}] = result
 	e[conversionPair{b, a}] = result
 	return result
 }
 
-func (e equalMemoryTypes) equal(a, b *types.Type) bool {
+func (e equalMemoryTypes) equal(a, b *types.Type, currentlyBuildingTypes map[*types.Type]bool) bool {
 	in, out := unwrapAlias(a), unwrapAlias(b)
 	switch {
 	case in == out:
 		return true
 	case in.Kind == out.Kind:
+		// if the type exists already, return early to avoid recursion
+		if currentlyBuildingTypes[in] {
+			return true
+		}
+		currentlyBuildingTypes[in] = true
```

Comment:
In Go passing map means effectively passing a pointer to map. Since noone is deleting anything from that map, you end up with map with a bunch of things at the end.

This probably isn't a big deal, but I think calling it "currentlyBuildingTypes" is a bit misleading then.